File: Medium/2026-04-04-2075-Decode-the-Slanted-Ciphertext/solution.py
class Solution:
    def decodeCiphertext(self, encodedText: str, rows: int) -> str:
        if rows == 0 or not encodedText:
            return ""
        
        n = len(encodedText)
        cols = (n + rows - 1) // rows
        matrix = [[' ' for _ in range(cols)] for _ in range(rows)]
        
        idx = 0
        for r in range(rows):
            for c in range(cols):
                if idx < n:
                    matrix[r][c] = encodedText[idx]
                    idx += 1
        
        originalText = []
        for diag in range(cols):
            r, c = 0, diag
            while r < rows and c >= 0:
                if matrix[r][c] != ' ':
                    originalText.append(matrix[r][c])
                r += 1
                c -= 1
        
        return ''.join(originalText).rstrip()

File: Medium/2026-04-04-2075-Decode-the-Slanted-Ciphertext/solution.java
class Solution {
    public String decodeCiphertext(String encodedText, int rows) {
        int n = encodedText.length();
        if (n == 0 || rows == 0) return "";
        
        int cols = (n + rows - 1) / rows; // Calculate the number of columns
        char[][] matrix = new char[rows][cols];
        
        // Fill the matrix with characters from encodedText
        for (int i = 0; i < n; i++) {
            int r = i % rows;
            int c = i / rows;
            if (c < cols) {
                matrix[r][c] = encodedText.charAt(i);
            }
        }
        
        StringBuilder originalText = new StringBuilder();
        
        // Read the matrix in the slanted order
        for (int diag = 0; diag < cols; diag++) {
            for (int r = 0; r < rows; r++) {
                int c = diag - r;
                if (c >= 0 && c < cols && matrix[r][c] != '\u0000') {
                    originalText.append(matrix[r][c]);
                }
            }
        }
        
        return originalText.toString().trim(); // Remove trailing spaces
    }
}